### PR TITLE
Gnz 1126 sdk npm package

### PR DESCRIPTION
**Pull request by stefandarius**
## Type of change

- [x] 🍕 New feature

## Description

[WIP]

The SDK will now be generated in a temp folder (user does not need to know its location). For `genezio local` we automatically perform a npm/yarn link in that location. The user than needs to run a command link `npm/yarn link @genezio/<project-name>` to get the sdk into their project.

The new sdk should work for both esm and cjs projects.

### CODE REVIEW

**Comment by vladiulianbogdan** [comment]
I got this error while I was testing it.

```
iulian-bogdanvlad@Iulian-Bogdans-MacBook-Pro mytest % genezio local --port 7777
[2023-10-11T17:57:49.256Z] DEBUG: [GenezioTelemetry] Europe/Bucharest GENEZIO_LOCAL INTERNAL-TEAM-SESSION en-US darwin undefined undefined
? It appears you don't have any SDK configured yet. Would you like to set one up now? Yes
? In which programming language is your frontend written? ts
? Which package manager are you using to install your frontend dependencies? npm
[2023-10-11T17:57:59.668Z] DEBUG: Start bundling...
[2023-10-11T17:57:59.678Z] DEBUG: [NodeJSBundler] Get the list of node modules and bundling the javascript code for file test.ts.
[2023-10-11T17:57:59.865Z] DEBUG: [NodeJSBundler] Copy non js files and node_modules for file test.ts.
[2023-10-11T17:57:59.884Z] DEBUG: [START_CLASS_PROCESS] Starting class MyTest on port 52887
[2023-10-11T17:57:59.884Z] DEBUG: [START_CLASS_PROCESS] Starting command: node
[2023-10-11T17:57:59.884Z] DEBUG: [START_CLASS_PROCESS] Parameters: /var/folders/w1/ky_fhr2d0vj4v3mg8w4vt3kh0000gn/T/genezio-93733/MyTest-6781445525739/local.mjs
Server listening on port 7777
[2023-10-11T17:57:59.929Z] DEBUG: Writing the SDK to files...
[2023-10-11T17:57:59.936Z] DEBUG: The SDK was successfully written to files.
Command failed: tsc -b tsconfig.esm.json tsconfig.cjs.json
internal/modules/cjs/loader.js:905
  throw err;
  ^

Error: Cannot find module '/Users/iulian-bogdanvlad/.nvm/versions/node/v14.20.1/pnpm-global/5/node_modules/typescript/bin/tsc'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:902:15)
    at Function.Module._load (internal/modules/cjs/loader.js:746:27)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:75:12)
    at internal/main/run_main_module.js:17:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}

iulian-bogdanvlad@Iulian-Bogdans-MacBook-Pro mytest %

**Review by radu1122 — APPROVED** [review]
looks good

File: src/commands/init.ts [hunk 1]
```diff
@@ -28,15 +35,24 @@ export async function initCommand(path:string) {
...
     if (!regions.includes(region)) {
-        log.error(red, `The region is invalid. Please use a valid region.\n Region list: ${regions}`);
+      log.error(
+        red,
+        `The region is invalid. Please use a valid region.\n Region list: ${regions}`,
+      );
     }
   }
 
-  GenezioTelemetry.sendEvent({eventType: TelemetryEventTypes.GENEZIO_INIT});
```
Comment (vladiulianbogdan) [inline src/commands/init.ts:39]: did this dissapear?

File: src/constants.ts [hunk 2]
```diff
@@ -11,6 +11,7 @@ let ENVIRONMENT: string;
 let SENTRY_DSN: string;
 
 const GENEZIO_TELEMETRY_ENDPOINT = "https://c4h2bia7gbokqdxxc6fe5sgj5e0imchy.lambda-url.us-east-1.on.aws/";
+const GENEZIO_REGISTRY = "5lraaxyjfn2jjgroseiwuzn3h40hmjxf.lambda-url.us-east-1.on.aws/RegistryHTTPHandler";
```
Comment (vladiulianbogdan) [inline src/constants.ts:14]: is this the same one for both dev and prod?

File: src/generateSdk/generateSdkApi.ts [hunk 3]
```diff
@@ -27,32 +34,42 @@ export async function sdkGeneratorApiHandler(projectConfiguration: YamlProjectCo
...
   }
 
   // Generate SDK if the yaml sdk field is present
-  if (!sdkGeneratorInput.sdk) {
-    return {
-      files: [],
-      sdkGeneratorInput: sdkGeneratorInput,
-    }
-  }
+  // if (!sdkGeneratorInput.sdk) {
```
Comment (vladiulianbogdan) [inline src/generateSdk/generateSdkApi.ts:63]: if we don't need this let's remove it

File: src/constants.ts (hunk 2 above)
Comment (stefandarius) [reply src/constants.ts:14]: initially i thought so, but i think we need to specify another one for prod. will leave as a placeholder for now

File: src/generateSdk/templates/packageJson.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
+export const getNodeModulePackageJsonLocal = (
```
Comment (vladiulianbogdan) [inline src/generateSdk/templates/packageJson.ts:1]: is this still used?

File: src/generateSdk/templates/packageJson.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+): string => `{
+  "name": "@genezio-sdk/${projectName}_${region}",
+  "version": "1.0.0",
+  "description": "",
+  "main": "./cjs/index.js",
+  "module": "./esm/index.js"
+}
+`;
+
+export const getNodeModulePackageJson = (
```
Comment (vladiulianbogdan) [inline src/generateSdk/templates/packageJson.ts:13]: same here

**Review by vladiulianbogdan — APPROVED** [review]

File: src/commands/init.ts (hunk 1 above)
Comment (stefandarius) [reply src/commands/init.ts:39]: it's back

**Pull request merged**